homeassistant/components/withings/common.py

```diff
@@ -530,7 +530,7 @@ async def async_get_data_manager(
     config_entry_data = hass.data[const.DOMAIN][config_entry.entry_id]
 
     if const.DATA_MANAGER not in config_entry_data:
-        profile: str = config_entry.data[const.PROFILE]
+        profile: str = config_entry.title
```

Comment:
We can't use the config entry title for any logic in the integration. The user can change the title. The title should only be used as a user facing name of the config entry. I think we should remove this property completely from the `DataManager`.

Reply:
It's used for the device name. Since it's the config entry title, we should be able to omit it. But For some reason, the device is named None even tho the mock config entry has title in it. This should work, I've seen it before, but it does not work. Any ideas?

Reply:
You removed the device name now.

Reply:
But a device without a name defaults to the config entry title

Reply:
Only primary devices. It's not a primary device anymore, but a link device.

Reply:
https://github.com/home-assistant/core/blob/38e013a90e3c0c6a3a739ff60efb291b969c8827/homeassistant/components/trafikverket_camera/entity.py#L25
Trafikverket camera also doesn't have a device name, but the tests still work since the device name is now the config entry title
https://github.com/home-assistant/core/blob/38e013a90e3c0c6a3a739ff60efb291b969c8827/tests/components/trafikverket_camera/test_binary_sensor.py#L19

I am expecting the same behaviour here, but I can't figure out why I don't see it

Reply:
What is the difference? Withings currently only exposes one device

Reply:
When you removed the name, you made the device entry a link device type.

https://github.com/home-assistant/core/blob/65c9e5ee13c35fd1ed8179ae140945e06d2c76f0/homeassistant/helpers/device_registry.py#L87-L116

Reply:
Aah, I did not know that. Thanks!